src/otx/core/data/utils/utils.py

```diff
@@ -133,23 +153,15 @@ def compute_robust_dataset_statistics(dataset: DatasetSubset, max_samples: int =
             )
 
     stat["annotation"]["num_per_image"] = compute_robust_statistics(np.array(num_per_images))
-    # The reason why polygon is used prior to others is based on assumtion that it is more accurate than other shapes.
-    # Especially, polygon can be used in the case both polygon and bbox exist like instance segmentation task.
-    # it's needed to refine this algorithm considering not only instance segmentation but also other tasks.
-    if "Polygon" in size_of_shapes:
-        stat["annotation"]["size_of_shape"] = compute_robust_scale_statistics(np.array(size_of_shapes["Polygon"]))
+    if task is OTXTaskType.INSTANCE_SEGMENTATION or task is OTXTaskType.SEMANTIC_SEGMENTATION:
+        stat["annotation"]["size_of_shape"] = compute_robust_scale_statistics(np.array(size_of_shapes[Polygon]))
+    elif task is OTXTaskType.DETECTION:
+        stat["annotation"]["size_of_shape"] = compute_robust_scale_statistics(
+            np.array(size_of_shapes[Bbox]),
+        )
     else:
-        max_ann_type = None
-        max_num_ann = 0
-        for ann_type, anns in size_of_shapes.items():
-            if max_num_ann < len(anns):
-                max_ann_type = ann_type
-                max_num_ann = len(anns)
-        if max_ann_type is not None:
-            stat["annotation"]["size_of_shape"] = compute_robust_scale_statistics(
-                np.array(size_of_shapes[max_ann_type]),
-            )
-
+        msg = f"Task type {task} is not supported."
+        raise NotImplementedError(msg)
```

Comment:
In my opinion, it's better to print warning and use existing annotation info even if task isn't one of det, iseg and sseg. It' because it should be more useful to use any annotation info to estimate dataset statistics than not using it even if it exists. For this reason, I think it's better to leave as it is, how do you think about it?

Reply:
Updated.